### DISCUSSION TITLE
After Successfull Deployment, there are missing JSON files

### DISCUSSION THREAD

**Discussion by DrazahSuzaku** [vercel/community#2089] [Help]
### Summary

After Successfull Deployment, my page have the 500 error. After reading the logs, i can see it's because a data JSON file is missing. This file is in the gitlab repository for the main branch and is supposed to be present. If i look in the output of the "source onglet" in Vercel, there is no trace of the folder who contain all the data JSON File.

I redeployed the application but no success :/

**Comment by MrMegaMango** [comment]
same thing here, I think the deployment drops "unrelated" files, which sucks!

**Comment by amyegan** [comment]
Hey there! Different frameworks and tools handle this in different ways. In many cases, files need to be put into specific directories to be in the final build. Can you tell us more about your project or share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)? That will help me get you more specific recommendations.

**Reply by ghost** [reply]
I saw you help someone with a similar issue by specifying the path with path.join(process.cwd(), dir, dir, filename) but when I tried your solution I still got 'file or dir not found' so I logged the filesystem and the directories that contain the json files are missing completely but in the vercel ui when I go into the Source tab in the nav bar menu 'Deployment, logs, functions, etc' and look at the file structure there the files are present its just after the build it seems those files vanish and my current file structure after process.cwd() is server/public/assets/filename.json Please let me know what you think I've been stuck on this for days feel free to checkout my public repo 
daregula/smart-settle
